app/src/pages/admin/serverSettingsPage/serverSettingsTabs/authConfigurationTab/forms/ssoUsersForm/ssoUsersForm.jsx

```diff
@@ -65,8 +82,27 @@ const SsoUsersFormComponent = ({ enabled: enabledFromStore, fetchAppInfo }) => {
   const getDescription = () =>
     formatMessage(enabled ? messages.ssoOnlyDescription : messages.manualInvitesDescription);
 
-  const handleToggle = (value) => {
-    setEnabled(value);
+  const handleToggle = async (value) => {
+    try {
+      await fetch('/api/v1/settings', {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        data: {
+          key: 'server.users.sso',
+          value: value.toString(),
+        },
+      });
+
+      await fetchAppInfo();
+      setEnabled(value);
```

Comment:
If you not rely on response from `fetchAppInfo`, you can set toggle as enabled right after the PUT request.
```suggestion
     setEnabled(value);
     await fetchAppInfo();
```

Reply:
Rewroto to a similar, bit slightly different pattern, implementing the same idea